From 025e4a0336d292f6767df0e5ef193bfab4ebaa55 Mon Sep 17 00:00:00 2001
From: deadman96385 <seanhoyt963@gmail.com>
Date: Wed, 21 Oct 2015 00:51:42 +0300
Subject: [PATCH 2/8] Open app when clicking on icon in App Info

Based on Xposed module Shortcut in App Info:
http://repo.xposed.info/module/com.mohammadag.shortcutinappinfo

Credit: Adam77Root
There is no vibration as in the module as I didn't find it necessary

Change-Id: I9c09a17bd0fd3a205ebb9cde366ce9d2335c6a84
[ jaysonedson@gmail.com: Ported to Pie/Android11 ]
Signed-off-by: Jason Edson <jaysonedson@gmail.com>
---
 .../settings/widget/EntityHeaderController.java   | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/src/com/android/settings/widget/EntityHeaderController.java b/src/com/android/settings/widget/EntityHeaderController.java
index 91a20b3..11363bc 100644
--- a/src/com/android/settings/widget/EntityHeaderController.java
+++ b/src/com/android/settings/widget/EntityHeaderController.java
@@ -24,6 +24,7 @@ import android.app.settings.SettingsEnums;
 import android.content.Context;
 import android.content.Intent;
 import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
 import android.graphics.drawable.ColorDrawable;
 import android.graphics.drawable.Drawable;
 import android.os.Bundle;
@@ -84,6 +85,7 @@ public class EntityHeaderController {
     // Required for hearing aid devices.
     private CharSequence mSecondSummary;
     private String mPackageName;
+    private String mPackageNameReal;
     private Intent mAppNotifPrefIntent;
     @UserIdInt
     private int mUid = UserHandle.USER_NULL;
@@ -148,6 +150,7 @@ public class EntityHeaderController {
      */
     public EntityHeaderController setIcon(ApplicationsState.AppEntry appEntry) {
         mIcon = Utils.getBadgedIcon(mAppContext, appEntry.info);
+        mPackageNameReal = appEntry.info.packageName;
         return this;
     }
 
@@ -249,6 +252,18 @@ public class EntityHeaderController {
         if (iconView != null) {
             iconView.setImageDrawable(mIcon);
             iconView.setContentDescription(mIconContentDescription);
+            if (mPackageNameReal != null) {
+                iconView.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        PackageManager pm = v.getContext().getPackageManager();
+                        Intent intent = pm.getLaunchIntentForPackage(mPackageNameReal);
+                        if (intent == null)
+                            return;
+                        v.getContext().startActivity(intent);
+                    }
+                });
+            }
         }
         setText(R.id.entity_header_title, mLabel);
         setText(R.id.entity_header_summary, mSummary);
-- 
2.31.1.606.gdf6c4f722c

